x-pack/winlogbeat/magefile.go

```diff
@@ -8,33 +8,78 @@
 package main
 
 import (
+	"context"
+	"fmt"
+
 	"github.com/magefile/mage/mg"
 
 	devtools "github.com/elastic/beats/v7/dev-tools/mage"
+	"github.com/elastic/beats/v7/dev-tools/mage/target/test"
 
-	// mage:import
+	//mage:import
 	_ "github.com/elastic/beats/v7/dev-tools/mage/target/common"
-	// mage:import
+	//mage:import
 	_ "github.com/elastic/beats/v7/dev-tools/mage/target/build"
-	// mage:import
+	//mage:import
 	_ "github.com/elastic/beats/v7/dev-tools/mage/target/pkg"
-	// mage:import
+	//mage:import
 	_ "github.com/elastic/beats/v7/dev-tools/mage/target/dashboards"
-	// mage:import
-	_ "github.com/elastic/beats/v7/dev-tools/mage/target/test"
-	// mage:import
-	"github.com/elastic/beats/v7/dev-tools/mage/target/unittest"
-	// mage:import
+	//mage:import
 	winlogbeat "github.com/elastic/beats/v7/winlogbeat/scripts/mage"
 )
 
 func init() {
-	unittest.RegisterGoTestDeps(winlogbeat.Update.Fields)
-
 	winlogbeat.SelectLogic = devtools.XPackProject
 	devtools.BeatLicense = "Elastic License"
+
+	RegisterGoTestDeps(winlogbeat.Update.Fields)
+	test.RegisterDeps(UnitTest)
+}
+
+var goTestDeps, pythonTestDeps []interface{}
+
+// RegisterGoTestDeps registers dependencies of the GoUnitTest target.
+func RegisterGoTestDeps(deps ...interface{}) {
+	goTestDeps = append(goTestDeps, deps...)
+}
+
+// RegisterPythonTestDeps registers dependencies of the PythonUnitTest target.
+func RegisterPythonTestDeps(deps ...interface{}) {
+	pythonTestDeps = append(pythonTestDeps, deps...)
+}
+
+// UnitTest executes the unit tests (Go and Python).
+func UnitTest() {
+	mg.SerialDeps(GoUnitTest, PythonUnitTest)
 }
 
 // Update is an alias for update:all. This is a workaround for
 // https://github.com/magefile/mage/issues/217.
 func Update() { mg.Deps(winlogbeat.Update.All) }
+
+// GoUnitTest executes the Go unit tests.
+// Use TEST_COVERAGE=true to enable code coverage profiling.
+// Use RACE_DETECTOR=true to enable the race detector.
+func GoUnitTest(ctx context.Context) error {
+	mg.SerialCtxDeps(ctx, goTestDeps...)
+	args := devtools.DefaultGoTestUnitArgs()
+	args.ExtraFlags = append(args.ExtraFlags, "-p", "1")
```

Comment:
Why `-p=1`? Might be worthy of a comment.

Reply:
Comment added. The rationale is sort of unsatisfying and may be worth looking into in the future, but for the moment this is the simpler solution. Quoting the comment here for convenience:
```
	// The module unit tests depend on a running docker container to provide
	// the ES instance to run the processor pipeline. In the absence of a
	// test supervisor or a single test executable to ensure that only a
	// single container is running, or additional logic to ensure no network
	// collisions, we ensure that only one test package is running at a time.
```